helix-term/src/ui/statusline.rs

```diff
@@ -179,16 +189,30 @@ where
                 "   "
             }
         ),
-        if visible && config.color_modes {
-            match context.editor.mode() {
-                Mode::Insert => Some(context.editor.theme.get("ui.statusline.insert")),
-                Mode::Select => Some(context.editor.theme.get("ui.statusline.select")),
-                Mode::Normal => Some(context.editor.theme.get("ui.statusline.normal")),
-            }
-        } else {
-            None
-        },
+        mode_style,
     );
+
+    match config.statusline.mode_separator {
+        ModeSeparator::Flat => {}
+        _ => {
+            // use mode style as mode separator style except set
+            // background to statusline background
+            let mode_separator_style = mode_style.map(|s| Style {
+                fg: s.bg,
+                bg: context.editor.theme.get("ui.statusline").bg,
+                ..Default::default()
+            });
+
+            let mode_separator = match config.statusline.mode_separator {
+                ModeSeparator::Angled => "",
+                ModeSeparator::Slanted => "",
+                ModeSeparator::Round => "",
+                ModeSeparator::Flat => unreachable!(),
+            };
```

Comment:
Like the current separator, this can just be specified as an arbitrary string rather than presets

Reply:
I agree but I also think the presents are very reasonable and most 90% of people would be using one of the presents. I think it would be best to do a hybrid approach that lets arbitrary strings be used but also provides presets. See commit 1c2b828 for what this might look like.

Reply:
These presets require a patched font to work (they're in the private use area of unicode) which is not acceptable for core. See https://github.com/helix-editor/helix/pull/2211#issuecomment-1106015472, https://github.com/helix-editor/helix/pull/2377#issuecomment-1128819954. A free-form string allows these without building it in to core.